src/java/org/apache/cassandra/cql3/constraints/ScalarColumnConstraint.java

```diff
@@ -0,0 +1,168 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.cql3.constraints;
+
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+
+import org.apache.cassandra.cql3.ColumnIdentifier;
+import org.apache.cassandra.cql3.CqlBuilder;
+import org.apache.cassandra.cql3.Operator;
+import org.apache.cassandra.db.TypeSizes;
+import org.apache.cassandra.db.marshal.AbstractType;
+import org.apache.cassandra.db.marshal.FloatType;
+import org.apache.cassandra.io.IVersionedSerializer;
+import org.apache.cassandra.io.util.DataInputPlus;
+import org.apache.cassandra.io.util.DataOutputPlus;
+import org.apache.cassandra.schema.ColumnMetadata;
+import org.apache.cassandra.utils.ByteBufferUtil;
+
+public class ScalarColumnConstraint implements ColumnConstraint<ScalarColumnConstraint>
+{
+    public final ColumnIdentifier param;
+    public final Operator relationType;
+    public final String term;
+
+    public final static Serializer serializer = new Serializer();
+
+    public final static class Raw
+    {
+        public final ColumnIdentifier param;
+        public final Operator relationType;
+        public final String term;
+
+        public Raw(ColumnIdentifier param, Operator relationType, String term)
+        {
+            this.param = param;
+            this.relationType = relationType;
+            this.term = term;
+        }
+
+        public ScalarColumnConstraint prepare()
+        {
+            return new ScalarColumnConstraint(param, relationType, term);
+        }
+    }
+
+    private ScalarColumnConstraint(ColumnIdentifier param, Operator relationType, String term)
+    {
+        this.param = param;
+        this.relationType = relationType;
+        this.term = term;
+    }
+
+    @Override
+    public void evaluate(Class<? extends AbstractType> valueType, Object columnValue)
+    {
+        Number columnValueNumber;
+        Double sizeConstraint;
+
+        try
+        {
+            columnValueNumber = (Number) columnValue;
+            sizeConstraint = Double.parseDouble(term);
```

Comment:
We're mixing treatment of the value between `Double` and `Float` in this file. Was that intentional and if so, why?